### PR TITLE
Backprop Test for Freeze FlaxWav2Vec2 Feature Encoder

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
This PR correctly implements a back propagation test to verify the functionality of the `freeze_feature_encoder` argument added to the FlaxWav2Vec2 Model in #15873. It tests:

1. That the computed loss for the frozen feature encoder model and unfrozen model are **equal**.
2. That the gradients of the frozen feature encoder **differ** to those of the unfrozen feature encoder.
3. That the gradients of all other unfrozen layers remain **equal**.

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
Great! Very nice test and good to see that no modeling code had to be changed

**Comment by sanchit-gandhi** [comment]
If @patil-suraj is happy with this test I'll merge!

File: tests/wav2vec2/test_modeling_flax_wav2vec2.py [hunk 1]
```diff
@@ -236,39 +237,65 @@ def test_freeze_feature_encoder(self):
...
+        self.assertEqual(grads.keys(), grads_frozen.keys())
+
+        # ensure that the gradients of the frozen layers differ, i.e. that the feature encoder is properly frozen
+        feature_extractor_grads = tuple(grads[k] for k in grads if "feature_extractor" in k)
+        feature_extractor_grads_frozen = tuple(grads_frozen[k] for k in grads_frozen if "feature_extractor" in k)
+
+        for feature_extractor_grad, feature_extractor_grad_frozen in zip(
+            feature_extractor_grads, feature_extractor_grads_frozen
+        ):
+            self.assert_difference(feature_extractor_grad, feature_extractor_grad_frozen, 1e-7)
```
Comment (patil-suraj) [inline tests/wav2vec2/test_modeling_flax_wav2vec2.py:283]: could we also add one more check to see if the grads of frozen module are all precisely zero, since that's what `jax.lax.stop_gradient` is supposed to do.

**Review by patil-suraj — APPROVED** [review]
Really nice tests, great job! Just left a comment.

File: tests/wav2vec2/test_modeling_flax_wav2vec2.py (hunk 1 above)
Comment (sanchit-gandhi) [reply tests/wav2vec2/test_modeling_flax_wav2vec2.py:283]: Thank you for the feedback! Sure thing, I'll look into that now!

Comment (sanchit-gandhi) [reply tests/wav2vec2/test_modeling_flax_wav2vec2.py]: The most recent commit [(`ca918e9`)](https://github.com/huggingface/transformers/pull/15938/commits/ca918e9c200e1d90734083632c7d66fc550b81e9) adds an assertion that verifies that the gradients of the frozen feature encoder layers are precisely zero!
